services/repository-managers/src/test/java/org/sagebionetworks/repo/manager/file/MultipartManagerV2ImplAutowireTest.java

```diff
@@ -128,17 +151,29 @@ public void testMultipartUpload() throws Exception {
 		assertNotNull(finalStatus.getResultFileHandleId());
 		fileHandlesToDelete.add(finalStatus.getResultFileHandleId());
 	}
-	
-	/**
-	 * Validation added for SYNPY-409 & PLFM-4183
-	 */
-	void validateUrl(String preSignedUrl) throws MalformedURLException {
-		URL url = new URL(preSignedUrl);
-		Map<String, String> map = Splitter.on('&').trimResults().withKeyValueSeparator("=").split(url.getQuery());
-		String expiresString = map.get("Expires");
-		assertNotNull("Expected the hacked 'Expires' parameter to be added to the URL for PLFM-4183", expiresString);
-		long expires = Long.parseLong(expiresString);
-		assertTrue("The hacked 'Expires' parameter should not be expired", (System.currentTimeMillis()/1000) < expires);
+
+	@Test
+	public void testMultipartUploadGoogleCloud() throws Exception {
+		Assume.assumeTrue(stackConfiguration.getGoogleCloudEnabled());
+		request.setStorageLocationId(googleCloudStorageLocationSetting.getStorageLocationId());
+		// step one start the upload.
+		MultipartUploadStatus status = startUpload();
+		String contentType = "application/octet-stream";
+		// step two get pre-signed URLs for the parts
+		String preSignedUrl = getPresignedURLForPart(status.getUploadId(), contentType);
+		// step three put the part to the URL
+		putStringToURL(preSignedUrl, fileDataString, contentType);
+		// step four add the part to the upload
+		addPart(status.getUploadId());
+		// Step five complete the upload
+		MultipartUploadStatus finalStatus = completeUpload(status.getUploadId());
+		// validate the results
+		assertNotNull(finalStatus);
+		assertEquals("1", finalStatus.getPartsState());
+		assertEquals(MultipartUploadState.COMPLETED, finalStatus.getState());
+		assertNotNull(finalStatus.getResultFileHandleId());
+		fileHandlesToDelete.add(finalStatus.getResultFileHandleId());
+

```

Comment:
Need some stress testing the simulates failures from google on large file uploads.